doc/scenarious.md

```diff
@@ -0,0 +1,126 @@
+
+# Handling Multiple Tables in the SQL Query Builder
+
+This document describes the possible scenarios of how the SQL Query Builder application interacts with multiple tables, including potential issues and approaches to solving them. The goal is to ensure correct interaction with the database in various scenarios and provide a logical and clear behavior for the user.
+
+---
+
+## Scenarios for Table Interaction
+
+### Scenario 1: Single Table
+- Description:
+  The user selects only one table, such as `Customers`.
+- Program Logic:
+  - The program displays a list of columns from the selected table.
+  - A simple SQL query is generated:
+    ```sql
+    SELECT CustomerID, CustomerName, ContactNumber
+    FROM Customers;
+    ```
+  - The user can add filters, sorting, and select specific columns.
+- Expected Result:
+  A table with data from the single selected table is displayed correctly.
+
+---
+
+### Scenario 2: Two Related Tables
+- Description:
+  The user selects two tables that have a clear relationship via a foreign key, such as `Customers` and `Orders` (related through `CustomerID`).
+- Program Logic:
+  - The program automatically identifies the relationship using the foreign key.
+  - By default, the program suggests an `INNER JOIN`. The user can manually change this to `LEFT JOIN`, `RIGHT JOIN`, or another type of join.
+  - If the user prefers, they can explicitly specify a `CROSS JOIN`.
+  - A query is generated:
+    ```sql
+    SELECT Customers.CustomerName, Orders.OrderID, Orders.OrderDate, Orders.TotalAmount
+    FROM Customers
+    INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID;
+    ```
+  - The user can select columns from both tables and add filters and sorting.
+- Expected Result:
+  A combined table with data from the two related tables is displayed correctly.
+
+---
+
+### Scenario 3: Two Unrelated Tables
+- Description:
+  The user selects two tables that do not have a clear relationship, such as `Customers` and `Products`.
```

Comment:
"clear" ...? before you mentioned "Two Unrelated Tables", not sure about not clear relationship between 2 tables BTW